src/settled-cash/CrossMarginCashMath.sol

```diff
@@ -356,45 +356,46 @@ library CrossMarginCashMath {
         //  default value of 0 is not a valid productId so the first run will always set the cache
         uint40 lastUsedProductId;
 
-        for (uint256 i; i < positions.length;) {
-            (, uint40 productId, uint64 expiry,,) = positions[i].tokenId.parseTokenId();
+        unchecked {
```

Comment:
I dont want to use `unchecked` because this will include all the math from `_processDetailWithToken`